skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java

```diff
@@ -0,0 +1,198 @@
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidationText;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDiagnosis;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDiagnosisList;
+import se.inera.intyg.common.support.modules.service.WebcertModuleService;
+
+public abstract class AbstractQuestionDiagnoser {
+
+    public static final short LIMIT_DIAGNOSIS_DESC = (short) 81;
+
+    protected static CertificateDataElement toCertificate(List<Diagnos> diagnoser, String questionId, String parentId, String textId,
+        String descriptionId, int index, CertificateTextProvider textProvider, String showValidationQuestionId,
+        String showValidationJsonId) {
+        return CertificateDataElement.builder()
+            .id(questionId)
+            .parent(parentId)
+            .index(index)
+            .config(
+                CertificateDataConfigDiagnoses.builder()
+                    .text(textProvider.get(textId))
+                    .description(descriptionId != null ? textProvider.get(descriptionId) : null)
+                    .terminology(
+                        List.of(
+                            DiagnosesTerminology.builder()
+                                .id(DIAGNOS_ICD_10_ID)
+                                .label(DIAGNOS_ICD_10_LABEL)
+                                .build(),
+                            DiagnosesTerminology.builder()
+                                .id(DIAGNOS_KSH_97_ID)
+                                .label(DIAGNOS_KSH_97_LABEL)
+                                .build()
+                        )
+                    )
+                    .list(
+                        List.of(
+                            DiagnosesListItem.builder()
+                                .id(DIAGNOSES_LIST_ITEM_1_ID)
+                                .build(),
+                            DiagnosesListItem.builder()
+                                .id(DIAGNOSES_LIST_ITEM_2_ID)
+                                .build(),
+                            DiagnosesListItem.builder()
+                                .id(DIAGNOSES_LIST_ITEM_3_ID)
+                                .build()
+                        )
+                    )
+                    .build()
+            )
+            .value(
+                CertificateDataValueDiagnosisList.builder()
+                    .list(createDiagnosValue(diagnoser))
+                    .build()
+            )
+            .validation(
+                new CertificateDataValidation[]{
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(questionId)
+                        .expression(withCitation(singleExpression(DIAGNOSES_LIST_ITEM_1_ID)))
+                        .build(),
+                    CertificateDataValidationText.builder()
+                        .limit(LIMIT_DIAGNOSIS_DESC)
```

Comment:
Does it not need id to be set?

Reply:
Its not needed, but I still think we should set id. Done.